samples/ServerSideBlazor.Server/Startup.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using ServerSideBlazor.Client;
+
+namespace ServerSideBlazor.Server
+{
+    public class Startup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Adds the Server-Side Blazor services, and those registered by the client startup.
+            services.AddServerSideBlazor<Client.Startup>();
+
+            // Since Blazor is running on the server, we can use an application service
+            // to read the forecast data.
+            services.AddSingleton<WeatherForecastService, DefaultWeatherForecastService>();
+        }
+
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            // Use component registrations from the client startup.
+            app.UseServerSideBlazor<Client.Startup>();
+
+            // Enables serving files from the client project.
+            app.UseBlazor<App>();
```

Comment:
It would be super cool if it's possible to merge these two (`UseServerSideBlazor` and `UseBlazor`), so the developer only needs to call `app.UseServerSideBlazor<Client.Startup>()`.

That is, to have `UseServerSideBlazor<T>` also call `UseBlazor<T>` internally. I *think* that would work fine because `UseBlazor<T>` only uses the `T` to identify the target project, so it would be file for it to be given the `Client.Startup` type for that generic param.

Then the developer is mostly just choosing between `UseBlazor` and `UseServerSideBlazor` depending on which hosting model they want. Of course the two methods may take different optional config info, but the symmetry would still feel really clean.

Are there any use cases where the developer would need to configure `UseBlazor` separately from `UseServerSideBlazor`? Is it maybe for the case where the developer wants *both* server and client-side hosting models together, with the final selection being made by which `<script>` tag they have in `index.html`?

Reply:
I can't really think of a case where you're doing server-side Blazor, but don't want to also serve files from that project. 

The only issues I can think of are more philosophical. If you called it multiple times with different startups you could get a big stack of file-serving middleware.  If you wanted to control the position of the various middleware, you can't since it's a single call. These are some of the reasons why we don't typically put multiple middleware behind a single method call. 

But that can be an issue for future us.